app/src/main/java/com/realityexpander/tasky/presentation/login_screen/LoginViewModel.kt

```diff
@@ -146,19 +170,35 @@ class LoginViewModel @Inject constructor(
             }
             is LoginEvent.IsValidEmail -> {
                 _loginState.value = _loginState.value.copy(
-                    isInvalidEmail = !event.isValid
+                    isInvalidEmail = !event.isValid,
+                )
+            }
+            is LoginEvent.ShowInvalidEmailMessage -> {
+                _loginState.value = _loginState.value.copy(
+                    isShowInvalidEmailMessage = true
                 )
```

Comment:
In general, I'd always use `_stateFlow.update { it.copy(...) }` to avoid race conditions. This is how it could happen:

*StateFlow holds value A*
- Thread 1 calls `copy()` on StateFlow with value A
- Before Thread 1 assigns its changed copy, Thread 2 updates StateFlow with value B
- Now, Thread 1 assigns its changed copy, which is still based on value A though, but the latest value is already value B. Therefore, value B would never be considered here.

Not always an issue, but I worked on highly stateful apps with frequently changing states, in which this really bit me, since such bugs are super hard to find.

Reply:
Makes sense, and I will make this change.